File: EPI/12.5/swift/12.5.swift
public func root(_ value: Int) -> Int {
    var low = 0
    var high = value

    while low <= high {
        let mid = low + (high - low) / 2

        if Double(mid) * Double(mid) <= Double(value) {
            low = mid + 1
        } else {
            high = mid - 1
        }
    }

    return low - 1
}

File: EPI/Searching/Computer the integer square root/typescript/12.5.ts
export default function sqrt(n: number): number {
    let low = 0;
    let high = n;

    while (low < high) {
        let mid = Math.ceil(low + (high - low) / 2);

        if (mid * mid > n) {
            high = mid - 1;
        } else {
            low = mid;
        }
    }

    return high;
}
